### PR TITLE
Add IST0158 message description

### PR DESCRIPTION
**Pull request by hanxiaop**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/reference/config/analysis/ist0158/index.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+layout: analysis-message
+owner: istio/wg-user-experience-maintainers
+test: n/a
+---
+
+This message is similar to [IST0105](/docs/reference/config/analysis/ist0105/), which is a warning message for a single pod.
+
+This message occurs when the namespace is enabled for automatic sidecar injection, but some pods in the namespace are not having the correct sidecar injected.
+
+There are several possible causes for the pod:
```
Comment (kfaseela) [inline content/en/docs/reference/config/analysis/ist0158/index.md:12]:
```suggestion
There are several possible causes for this:
```

File: content/en/docs/reference/config/analysis/ist0158/index.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+* Automatic sidecar injection is enabled (default enabled unless explicitly
```
Comment (kfaseela) [inline content/en/docs/reference/config/analysis/ist0158/index.md:14]: not sure how this is a cause for the warning message?

File: content/en/docs/reference/config/analysis/ist0158/index.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  disabled during installation.)
+* The pod is running in a namespace where sidecar injection is enabled (the
+  namespace has the label `istio-injection=enabled`)
+* The proxy version running on the sidecar does not match the version used by
+  the auto-injector
+
+This often results after upgrading the Istio control plane; after upgrading
```
Comment (kfaseela) [inline content/en/docs/reference/config/analysis/ist0158/index.md:21]:
```suggestion
This often happens as a result of upgrading the Istio control plane; after upgrading
```

File: content/en/docs/reference/config/analysis/ist0158/index.md (hunk 1 above)
Comment (hanxiaop) [reply content/en/docs/reference/config/analysis/ist0158/index.md:12]: I just copy&paste some of the descriptions from the `IST0105` message, will revise some of the descriptions a little bit

File: content/en/docs/reference/config/analysis/ist0158/index.md (hunk 2 above)
Comment (hanxiaop) [reply content/en/docs/reference/config/analysis/ist0158/index.md:14]: Will revise to make situations more clear

File: content/en/docs/reference/config/analysis/ist0158/index.md (hunk 3 above)
Comment (hanxiaop) [reply content/en/docs/reference/config/analysis/ist0158/index.md]: Fixed

**Review by kfaseela — APPROVED** [review]

**Pull request merged**